### ISSUE TITLE
Setting auto-scaler to min-count 0 will not scale up windows nodes

### ISSUE THREAD

**Issue by jeanfrancoislarente** [Azure/AKS#1882]
**What happened**:

I've created a cluster with a system pool
```
az aks nodepool add `
    --resource-group $resourceGroup `
    --cluster-name $resourceGroup `
    --name system `
    --node-count 2 `
    --node-vm-size "Standard_E2s_v3" `
    --mode System `
    --node-taints CriticalAddonsOnly=true:NoSchedule
```

one linux user pool

```
az aks nodepool add `
    --resource-group $resourceGroup `
    --cluster-name $resourceGroup `
    --name linux `
    --node-count 0 `
    --min-count 0 `
    --max-count 5 `
    --node-vm-size "Standard_E16s_v3" `
    --max-pods 250 `
    --enable-cluster-autoscaler
```

and one windows user pool

```
az aks nodepool add `
    --resource-group $resourceGroup `
    --cluster-name $resourceGroup `
    --name win `
    --os-type Windows `
    --node-count 0 `
    --min-count 0 `
    --max-count 5 `
    --node-vm-size "Standard_E16s_v3" `
    --max-pods 250 `
    --enable-cluster-autoscaler
```

**What you expected to happen**:

- node count should be zero if no deployments
- deployment should trigger a scale up of windows nodes 0 -> 1

**How to reproduce it (as minimally and precisely as possible)**:

Create a cluster with the above commands (I create the cluster with the default nodepool, added the system node pool and deleted the original, since `--node-taints CriticalAddonsOnly=true:NoSchedule` is not supported on `az aks create`

Try to deploy something with nodeSelector `kubernetes.io/os=windows`

**Anything else we need to know?**:

**Environment**:
Reproduced in 1.17.9, 1.18.8 and 1.19.1

fyi @palma21

**Comment by palma21** [comment]
We looked into the issue and found it. With the change from beta to stable on the selectors some haven't transitioned when CA is predicting the template (since there's no existing node to base itself on, it predicts).

If you use `beta.kubernetes.io/os=windows` this should work, and we'll correct this to make sure all the new built-in labels are captured.

**Comment by jeanfrancoislarente** [comment]
To clarify, I don't need to add labels to the nodepool, I need to change my selector to beta.kubernetes.io/os=windows ?

**Comment by marwanad** [comment]
@jeanfrancoislarente yup that should work in your case. If your pod nodeSelector was on the non-beta OS label.

**Comment by jeanfrancoislarente** [comment]
Thanks @marwanad! 

I can confirm the workaround of changing my nodeSelector to beta worked and scaled up my "zero-node" windows nodepool!

Any chance on an ETA for the permanent fix? Do I go through the trouble of pushing my new helm charts through or are we talking a couple of days? :)

**Comment by marwanad** [comment]
@jeanfrancoislarente Not ideal but it would be with next week's release cycle(so ETA in all regions would be 1.5 weeks from that). Alternatively, but a bit more work is to disable CA, manually scale the windows pool to one node. Then re-enable CA with minimum count set to 0. It should pick-up on the correct templates in that case.

Obviously, modifying your helm charts would be the better short-term mitigation for that.

**Comment by jeanfrancoislarente** [comment]
I've updated our charts for now, easy search/replace. Thanks!

**Comment by marwanad** [comment]
Fixe has been rolled out. Feel free to comment on the issue if you're still hitting it.
